nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -50,6 +54,10 @@
 
     public static final String COLD_STORAGE_BEING_RETRIEVED_PROPERTY = "coldstorage:beingRetrieved";
 
+    public static final String GET_DOCUMENTS_TO_CHECK_QUERY = "SELECT * FROM Document WHERE coldstorage:beingRetrieved = 1";
```

Comment:
You'll probably want to add an index for this in `MongoDBRepository` (until the day where we have a more configurable way to do this).

Reply:
👍 and what about the VCS repository you think we should add it on `postgresSql`

Reply:
Yes probably. In `SQLInfo.initHierarchySQL`

Reply:
for VCS i think i will create a new method as it is done [here](https://github.com/nuxeo/nuxeo/blob/master/nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/SQLInfo.java#L845) because we will have a new table named `coldstorage` as we create a new specific `xsd` to hold the coldstorage information if your agree i will add this new method and i will create a new index on `beingretrieved` column

Reply:
as discussed on Slack we will not add an index on `VCS`, only on Mongo. If needed the customer will add it on PostgresSQL... directly on the DB